### ISSUE TITLE
Feature: Convert SVG via URL parameter

### ISSUE THREAD

**Issue by shshaw** [jakearchibald/svgomg#26]
It would be awesome if SVGO could optimize SVG markup provided by a URL parameter ( `https://jakearchibald.github.io/svgomg/?svg=%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2050%2050%22%3E ... ` ). This would allow quick optimization from another webapp or even from a desktop app like [Alfred](http://www.alfredapp.com/).

**Comment by maban** [comment]
I would love this too, and was thinking of raising a PR to implement it (if I can get it to work). @jakearchibald Is this something you think would be useful, too?

**Comment by jerone** [comment]
This could result in very long urls. Remember that there are length [limits](http://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers) to browsers and servers can handle.

**Comment by jakearchibald** [comment]
Putting the SVG in the URL fragment would avoid it being sent to the server.

**Comment by shshaw** [comment]
A hash? `https://jakearchibald.github.io/svgomg/#%3Csvg...`

**Comment by jakearchibald** [comment]
Yeah I reckon it'd be useful. Might not even be too difficult (famous last words I know), the loading code is abstracted from the loading method.

**Comment by shshaw** [comment]
Yes, that would protect your server as well. The URL limits may still apply, but it'll depend on the browser.

**Comment by jakearchibald** [comment]
https://github.com/jakearchibald/svgomg/blob/master/src/js/page/main-controller.js calling onInputChange with {data, filename} should do it.
